delayed.py

```diff
@@ -73,8 +73,23 @@ def host_info_obtainer(conn, vc):
                 new_host_info = data.HostRuntimeInfo(**item)
                 new_host_info.created_at = datetime.datetime.now()
                 new_host_info.save(conn=conn)
-        logger.info(f"host_info_obtainer finished successfully " +
-                    f"in: {time.time() - start_host_info_obtainer}")
+
+        # hosts that are currently not present in vmware folder must be deleted from db as well
+        host_names = [item["name"] for item in info]
+        hosts_to_be_deleted = []
+        ids_to_be_deleted = []
+        for hostinfo in data.HostRuntimeInfo.get({}, conn=conn):
+            if hostinfo.name not in host_names:
+                hosts_to_be_deleted.append(hostinfo.name)
+        for host_name in hosts_to_be_deleted:
+            host = data.HostRuntimeInfo.get({'name': host_name}, conn=conn)
+            ids_to_be_deleted.append(host[0].id)
+        for host_id in ids_to_be_deleted:
+            data.HostRuntimeInfo.delete({'_id': host_id}, conn=conn)
+            logger.debug(f'host: {host_id} deleted from database')
+
+        logger.info(f'host_info_obtainer finished successfully ' +
+                    f'in: {time.time() - start_host_info_obtainer}')
```

Comment:
Apologies, but the function is called "host_info_obrainer" and this part of code performs data deletion - I would extract it to a separate method.

Reply:
it may be done

Reply:
ok but array `info` should be returned from the original function and given to the next function in the chain as a param?

Reply:
I've added something but result from current code is needed!